libqtile/scripts/migrations/rename_unspecified.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2023, elParaguayo. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from itertools import filterfalse
+
+import libcst as cst
+import libcst.matchers as m
+
+from libqtile.scripts.migrations._base import (
+    Check,
+    MigrationTransformer,
+    _QtileMigrator,
+    add_migration,
+)
+
+
+class RenameUnspecifiedTransformer(MigrationTransformer):
+    @m.leave(m.Call(func=m.Attribute(attr=m.Name("set_font"))))
+    def update_unspecified_arg(self, original_node, updated_node) -> cst.Call:
+        args = original_node.args
+
+        def is_interesting_arg(arg):
+            return m.matches(arg.value, m.Name("UNSPECIFIED"))
+
+        args = list(filterfalse(is_interesting_arg, original_node.args))
+
+        if len(args) == len(original_node.args):
+            return original_node
+
+        return updated_node.with_changes(args=args)
+
+    def strip_unspecified_imports(self, original_node, updated_node) -> cst.Import:
+        self.lint(original_node, "'UNSPECIFIED' can be dropped")
```

Comment:
Isn't this going to be added even if `UNSPECIFIED` isn't in the import line? We only want the linting if it's found.

The tests only check if linting is present if there's a change. Should we change that so it also checks that there's no linting if there's no change? In that case, a `NoChange` test here with a standard import line would need to provide no linting.

Reply:
Yes, good point. I moved the `self.lint` call, I can add a no-lint-if-no-changes test in another PR.